packages/form-state-manager/src/files/compose-validators.ts

```diff
@@ -10,15 +10,15 @@ import ComposeValidators from '../types/compose-validators';
  * @returns {Function} New validation function
  */
 const composeValidators: ComposeValidators = (validators = []) => (value, allValues) => {
-  const promises: Promise<string | undefined>[] = [];
+  const promises: Promise<any>[] = [];
```

Comment:
This should not be any. It should be `string | undefined | WarningResult` or something.

Reply:
I think it can - what about react nodes for example?

Reply:
We must have a strict declaration of the error message format. Anybody can make nodes from it in mapper. We should not expect that API will return react node...

Reply:
This is a function that is used by the state manager and not the react binding for it. Imagine that the manager is not used withing react context. How would they consume nodes?

Reply:
I am just saying that error can be now anything and it works - see the example below:

see https://stackblitz.com/edit/5q2uq8?file=index.js

So I don't see any reason why to limit it when there is no computation/manipulation (so it cannot be break if you put there anything) with this value.

Reply:
Ofc it can be anything. I will rephrase the question. What is in our docs? Validators should return `undefined` or `string` or a warning object. Why do we expect anything? Do we remove that rule?

Reply:
If you want to expand on our implementation, make the validator a generic type and make it extend the return type with a user-defined type. That way we can keep it strictly and safely defined.

Reply:
Or at minimum make it `string | undefined | WarningResult | any`.

Reply:
https://data-driven-forms.org/schema/custom-validator

There is already `any`.

Real example: In sources,  `<FormattedMessage />` is put there.

Reply:
Then it is wrong. And it should be fixed. `WarningResult` is an important structure because it cannot be used as a node and will crash in runtime.

Reply:
I don't understand. What will crash in runtime?

Reply:
I meant compile-time sorry. The compose validator is an exported function. Yes, we do catch the warning object internally and assing the warning to the meta-object, but if used outside it can return an object that is not usable for render. When using TS, you will get an error that the return type is any and the code will not compile.

By changing the function type to a generic function, you will avoid additional typecasting and let the user define return type of their function. Therefore no compile time errors and typecasting:
```ts
type defaultType = string | warning | any
function ComposeValidators<T = defaultType>(arg: foo): T {
  return arg;
}
```